src/main.rs

```diff
@@ -1,3 +1,131 @@
-fn main() {
-    println!("Hello, world!");
+use crate::config::read_config;
+use anyhow::Context;
+use hickory_proto::xfer::DnsResponse;
+
+use std::str::FromStr;
+use time::Duration;
+
+use hickory_client::{
+    client::Client,
+    op::ResponseCode,
+    rr::{rdata::TXT, Name, RData, Record, RecordType},
+};
+
+use log::{debug, info, trace};
+extern crate pretty_env_logger;
+#[macro_use]
+extern crate log;
```

Comment:
I'd suggest using `tracing` and `tracing_subscriber` instead. The macros use the same format, and it's backwards compatible with `log`, but it offers a lot more when it comes to structured logging and spans and such things.